engines/e_devcrypto.c

```diff
@@ -1172,17 +1219,29 @@ static int devcrypto_ctrl(ENGINE *e, int cmd, long i, void *p, void (*f) (void))
  */
 static int open_devcrypto(void)
 {
+    int fd;
+
     if (cfd >= 0)
         return 1;
 
-    if ((cfd = open("/dev/crypto", O_RDWR, 0)) < 0) {
+    if ((fd = open("/dev/crypto", O_RDWR, 0)) < 0) {
 #ifndef ENGINE_DEVCRYPTO_DEBUG
         if (errno != ENOENT)
 #endif
             fprintf(stderr, "Could not open /dev/crypto: %s\n", strerror(errno));
         return 0;
     }
 
+#ifdef CRIOGET
+    if (ioctl(fd, CRIOGET, &cfd) < 0) {
+        fprintf(stderr, "Could not create crypto fd: %s\n", strerror(errno));
+        cfd = -1;
+        return 0;
+    }
+#else
+    cfd = fd;
```

Comment:
I will not look too closely at the thread-safety of updates to `cfd`, since that is out of scope of *this* PR...

Reply:
Yes, those are old bugs. :(  Are engines supposed to be thread safe?  There are also races with concurrent commands to devcrypto_ctrl()

Reply:
Part of why I was not going to look closely is that I don't remember what the rules are, offhand.  I expect that someone like @levitte would have a better sense of them, but many folks are out of the office this week so I wasn't expecting a quick answer on that part.

Reply:
While working on a 1.1.1 PR, I noticed that I leaked `fd` in the `CRIOGET` case. :(  I will test a fix for that and open a PR for that first and will include it in the 1.1.1 PR once it's been merged.